core/translate/delete.rs

```diff
@@ -0,0 +1,139 @@
+use std::rc::Weak;
+use std::{cell::RefCell, rc::Rc};
+
+use sqlite3_parser::ast::{self, QualifiedName, With, Operator};
+
+use crate::{
+    schema::{Schema, Table},
+    storage::sqlite3_ondisk::DatabaseHeader,
+    translate::expr::{translate_expr}, 
+    translate::planner::bind_column_references,
+    vdbe::{builder::ProgramBuilder, Insn, Program},
+};
+use crate::{Connection, Result};
+
+use super::plan::BTreeTableReference;
+
+pub fn translate_delete(
+    schema: &Schema,
+    with: &Option<With>,
+    qualified_table_name: &QualifiedName, 
+    where_clause: &Option<Box<ast::Expr>>,
+    database_header: Rc<RefCell<DatabaseHeader>>,
+    connection: Weak<Connection>,
+) -> Result<Program> {
+    assert!(with.is_none()); // Don't handle WITH clause yet
+
+    let mut program = ProgramBuilder::new();
+
+    let init_label = program.allocate_label();
+    program.emit_insn_with_label_dependency(
+        Insn::Init {
+            target_pc: init_label,
+        },
+        init_label,
+    );
+    let start_offset = program.offset();
+
+    let table_name = &qualified_table_name.name.0;
```

Comment:
need to use `normalize_ident()` for this to make sure e.g. both `"users"` and `[users]` match the table `users`